src/solvers/dg_tree/dg_1d.jl

```diff
@@ -88,37 +88,47 @@ function rhs!(du::AbstractArray{<:Any,3}, u, t,
   @timeit_debug timer() "reset ∂u/∂t" du .= zero(eltype(du))
 
   # Calculate volume integral
-  @timeit_debug timer() "volume integral" calc_volume_integral!(du, u, have_nonconservative_terms(equations), equations,
-                                                                dg.volume_integral, dg, cache)
+  @timeit_debug timer() "volume integral" calc_volume_integral!(
+    du, u, mesh,
+    have_nonconservative_terms(equations), equations,
+    dg.volume_integral, dg, cache)
 
   # Prolong solution to interfaces
-  @timeit_debug timer() "prolong2interfaces" prolong2interfaces!(cache, u, equations, dg)
+  @timeit_debug timer() "prolong2interfaces" prolong2interfaces!(
+    cache, u, mesh, equations, dg)
 
   # Calculate interface fluxes
-  @timeit_debug timer() "interface flux" calc_interface_flux!(cache.elements.surface_flux_values,
-                                                              have_nonconservative_terms(equations), equations,
-                                                              dg, cache)
+  @timeit_debug timer() "interface flux" calc_interface_flux!(
+    cache.elements.surface_flux_values, mesh,
+    have_nonconservative_terms(equations), equations,
+    dg, cache)
 
   # Prolong solution to boundaries
-  @timeit_debug timer() "prolong2boundaries" prolong2boundaries!(cache, u, equations, dg)
+  @timeit_debug timer() "prolong2boundaries" prolong2boundaries!(
+    cache, u, mesh, equations, dg)
 
   # Calculate boundary fluxes
-  @timeit_debug timer() "boundary flux" calc_boundary_flux!(cache, t, boundary_conditions, equations, dg)
+  @timeit_debug timer() "boundary flux" calc_boundary_flux!(
+    cache, t, boundary_conditions, mesh, equations, dg)
 
   # Calculate surface integrals
-  @timeit_debug timer() "surface integral" calc_surface_integral!(du, equations, dg, cache)
+  @timeit_debug timer() "surface integral" calc_surface_integral!(
+    du, mesh, equations, dg, cache)
 
   # Apply Jacobian from mapping to reference element
-  @timeit_debug timer() "Jacobian" apply_jacobian!(du, equations, dg, cache)
+  @timeit_debug timer() "Jacobian" apply_jacobian!(
+    du, mesh, equations, dg, cache)
 
   # Calculate source terms
-  @timeit_debug timer() "source terms" calc_sources!(du, u, t, source_terms, equations, dg, cache)
+  @timeit_debug timer() "source terms" calc_sources!(
+    du, u, t, source_terms, equations, dg, cache)
 
   return nothing
 end
 
 
 function calc_volume_integral!(du::AbstractArray{<:Any,3}, u,
```

Comment:
Since we are already cleaning up stuff...

Shouldn't we remove the `::AbstractArray{<:Any, <ndims> + 2}` annotations from `du` everywhere and instead move the dispatch-on-spatial-dimenesion to a more self-explanatory parameter (e.g., `equations::AbstractEquations{2}`)? This essentially applies to all 3 DG solvers and other methods than this.

But maybe the `AbstractArray` was here for a different reason that I forgot about...

Reply:
No, that's definitely right and also on the TODO list in #542. Shall we do it here or later in another PR?

Reply:
No, a separate PR sounds better!

Reply:
I did it (mostly) in e40b61ed

Reply:
Thanks a lot! I am not 100% convinced whether dispatching on the dimensionality of the mesh is the way to go, rather than the equations, but I guess time will tell...

Reply:
Yeah... But we dispatch on the mesh type anyways, so I thought that would just be easier for now. But that's not fixed forever, so we can change it if the need becomes obvious